pkg/operator2/operator.go

```diff
@@ -430,40 +435,46 @@ func (c *authOperator) checkReady(
 		return oauthClientsReady, fmt.Errorf("unable to check OAuth clients' readiness: %v", err)
 	}
 	if !oauthClientsReady {
-		c.setProgressingStatus(operatorConfig, "OAuthClientsNotReady", oauthClientsMsg)
+		setProgressingTrueAndAvailableFalse(operatorConfig, "OAuthClientNotReady", oauthClientsMsg)
 		return oauthClientsReady, nil
 	}
 
 	return true, nil
 }
 
-func (c *authOperator) checkDeploymentReady(deploymentVersionHash string) (bool, string, error) {
+func (c *authOperator) checkDeploymentReady(deploymentVersionHash string, operatorConfig *operatorv1.Authentication) (bool, error) {
+	reason := "OAuthServerDeploymentNotReady"
 	deployments := c.deployments.Deployments(targetName)
 	osinDeployment, err := deployments.Get(targetName, metav1.GetOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
-			return false, "deployment does not exist", nil
+			setProgressingTrueAndAvailableFalse(operatorConfig, reason, "deployment does not exist")
+			return false, nil
 		}
-		return false, "", err
+		return false, err
 	}
 
 	if osinDeployment.ObjectMeta.Annotations[deploymentVersionHashKey] != deploymentVersionHash {
-		return false, "deployment does not yet have the expected version", nil
+		setProgressingTrue(operatorConfig, reason, "deployment does not yet have the expected version")
+		return false, nil
 	}
 
 	if osinDeployment.ObjectMeta.Generation != osinDeployment.Status.ObservedGeneration {
-		return false, "deployment's observed generation did not reach the expected generation", nil
+		setProgressingTrue(operatorConfig, reason, "deployment's observed generation did not reach the expected generation")
+		return false, nil
 	}
 
 	if osinDeployment.DeletionTimestamp != nil {
-		return false, "", fmt.Errorf("the deployment is being deleted")
+		setProgressingTrueAndAvailableFalse(operatorConfig, reason, "deployment is being deleted")
+		return false, nil
 	}
 
 	if osinDeployment.Status.UpdatedReplicas != osinDeployment.Status.Replicas || osinDeployment.Status.UnavailableReplicas > 0 {
-		return false, "not all deployment replicas are ready", nil
+		setProgressingTrue(operatorConfig, reason, "not all deployment replicas are ready")
```

Comment:
We are available here as long as we have one ready replica.

Reply:
right, updated, missed that!